packages/core/types/src/types/core/common/api.ts

```diff
@@ -1,3 +1,87 @@
 import type { Common } from '..';
 
 export interface API extends Common.Module {}
+
+export interface HttpError<TDetails = unknown> {
+  status: number;
+  name: string;
+  message: string;
+  details: TDetails;
+}
+
+export interface BadRequestError<TDetails = unknown> extends HttpError<TDetails> {
```

Comment:
@innerdvations  I added a list of all errors we can throw from the server side in strapi/types, But I was unsure if this was the right place. Do you know of any other place we could fit this?

Having this list of errors really helps defining the api contracts

Reply:
I know the classes exists, but do those types can help the client with type guarding? Rather than the base structure of what a error contains, they can not be used for that as far as I know. 
but hapy to remove duplicated stuff if it doesnt add much value!

Reply:
Maybe I can look for a way to not duplicate it as much, and leave a much simpler type than this 🤔

Reply:
The classes can be used as types I believe, we've used them here:

https://github.com/strapi/strapi/blob/160ac2f527cf0d67125a47414886992352b02970/packages/core/helper-plugin/src/utils/normalizeAPIError.ts#L46

Is this what you were thinking? Or were you thinking of something more? :)

Reply:
what I mean is you need to do this
errors.YupValidationError & { details: { errors: YupFormattedError[] } } 

Wether with this approach you would not need to extend the type

Reply:
ahhhhhhhhhhhhhhh I think it'd be nice for sure. You might be able to do it by changing the types in the utils folder? I didn't try at the moment 🤓 but just doing `instanceOf YupValidationError` and knowing what the details are would be v helpful assuming it's uniform across said errors

Reply:
the only problem I found with that was modifying those, in some cases we dynamically modify the message and also the details. And I didnt find a way to go around that with the Error classes :( But will try a bit more just in case, i might be nice to reuse them

Reply:
I think you were right josh, I will try to add the same generics into the strapi utils error classes, and that should work the same without having the duplicate definitions